##                      _             _ 
##  _ __ ___   ___   __| |    ___ ___| |  mod_ssl
## | '_ ` _ \ / _ \ / _` |   / __/ __| |  Apache Interface to OpenSSL
## | | | | | | (_) | (_| |   \__ \__ \ |  www.modssl.org
## |_| |_| |_|\___/ \__,_|___|___/___/_|  ftp.modssl.org
##                      |_____|         
## ____________________________________________________________________________
##
## Annotated patch file: addon.patch
## Copyright (c) 1998-2001 Ralf S. Engelschall, All Rights Reserved. 
## Created on: 18-Jul-2003
##
## This file assembles changes to existing Apache source files
## between the original Apache and the patched variant. It can be
## automatically applied to a vanilla Apache source tree with the
## 'patch' tool to upgrade those files.  Each patch snippet is
## annotated with a short description.
##

+---------------------------------------------------------------------------
| Add an entry for mod_define.
+---------------------------------------------------------------------------
Index: src/Configuration.tmpl
--- src/Configuration.tmpl	28 Jan 2002 19:21:21 -0000	1.1.1.7
+++ src/Configuration.tmpl	28 Jan 2002 19:40:56 -0000	1.23
@@ -258,6 +313,11 @@
 
 AddModule modules/standard/mod_env.o
 
+## mod_define expands variables on arbitrary directive lines.
+## It requires Extended API (EAPI).
+
+# AddModule modules/extra/mod_define.o
+
 ##
 ## Request logging modules
 ##

+---------------------------------------------------------------------------
| Add more beautiful optic to the status page table..
+---------------------------------------------------------------------------
Index: src/modules/standard/mod_status.c
--- src/modules/standard/mod_status.c	18 Jul 2003 15:44:39 -0000	1.1.1.15
+++ src/modules/standard/mod_status.c	18 Jul 2003 15:56:59 -0000	1.12
@@ -484,12 +484,33 @@
 	    if (no_table_report)
 		ap_rputs("<p><hr><h2>Server Details</h2>\n\n", r);
 	    else
+#ifndef NO_PRETTYPRINT
+		ap_rputs("<p>\n\n<table bgcolor=\"#ffffff\" border=\"0\">"
+			"<tr bgcolor=000000>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Srv</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>PID</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Acc</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>M</b></font></td>"
+#ifndef NO_TIMES
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>CPU</b></font></td>"
+#endif
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>SS</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Req</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Conn</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Child</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Slot</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Host</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>VHost</b></font></td>"
+			"<td><font face=\"Arial,Helvetica\" color=\"#ffffff\"><b>Request</b></td>"
+			"</tr>\n", r);      
+#else /* NO_PRETTYPRINT */
 #ifdef NO_TIMES
 		/* Allow for OS/2 not having CPU stats */
 		ap_rputs("<p>\n\n<table border=0><tr><th>Srv<th>PID<th>Acc<th>M\n<th>SS<th>Req<th>Conn<th>Child<th>Slot<th>Client<th>VHost<th>Request</tr>\n\n", r);
 #else
 		ap_rputs("<p>\n\n<table border=0><tr><th>Srv<th>PID<th>Acc<th>M<th>CPU\n<th>SS<th>Req<th>Conn<th>Child<th>Slot<th>Client<th>VHost<th>Request</tr>\n\n", r);
 #endif
+#endif /* NO_PRETTYPRINT */
 	}
 
 	for (i = 0; i < HARD_SERVER_LIMIT; ++i) {
@@ -617,6 +638,11 @@
 				vhost->server_hostname) : "(unavailable)");
 		    }
 		    else {		/* !no_table_report */
+#ifndef NO_PRETTYPRINT
+			ap_rprintf(r,"<tr bgcolor=\"#ffffff\">");
+#else
+			ap_rprintf(r,"<tr>");
+#endif
 			if (score_record.status == SERVER_DEAD)
 #ifdef TPF
                             if (kill(ps_record.pid, 0) == 0) {
@@ -629,12 +655,12 @@
                             } else
 #endif /* TPF */
 			    ap_rprintf(r,
-				"<tr><td><b>%d-%d</b><td>-<td>%d/%lu/%lu",
+				"<td><b>%d-%d</b><td>-<td>%d/%lu/%lu",
 				i, (int) ps_record.generation,
 				(int) conn_lres, my_lres, lres);
 			else
 			    ap_rprintf(r,
-				"<tr><td><b>%d-%d</b><td>%d<td>%d/%lu/%lu",
+				"<td><b>%d-%d</b><td>%d<td>%d/%lu/%lu",
 				i, (int) ps_record.generation,
 				(int) ps_record.pid, (int) conn_lres,
 				my_lres, lres);
@@ -737,6 +774,12 @@
 </table>\n", r);
 #endif
 	}
+
+#ifdef EAPI
+    ap_hook_use("ap::mod_status::display",
+                AP_HOOK_SIG4(void,ptr,int,int), AP_HOOK_ALL,
+                r, no_table_report, short_report);
+#endif
 
     } else {
 

+---------------------------------------------------------------------------
| Add a hyperlink to the mod_define.html document.
+---------------------------------------------------------------------------
Index: htdocs/manual/mod/index.html.en
--- htdocs/manual/mod/index.html.en	28 Jan 2002 19:21:43 -0000	1.1.1.2
+++ htdocs/manual/mod/index.html.en	28 Jan 2002 19:40:56 -0000	1.3
@@ -101,6 +101,10 @@
       <dd>Support for Netscape-like cookies. Replaced in Apache 1.2
       by mod_usertrack</dd>
 
+      <dt><a href="mod_define.html">mod_define</a></dt>
+
+      <dd>Variable Definition for Arbitrary Directives</dd>
+
       <dt><a href="mod_digest.html">mod_digest</a> Apache 1.1 and
       up</dt>
 

+---------------------------------------------------------------------------
| Add a hyperlink for the Define directives.
+---------------------------------------------------------------------------
Index: htdocs/manual/mod/directives.html.en
--- htdocs/manual/mod/directives.html.en	18 Jul 2003 15:44:16 -0000	1.1.1.5
+++ htdocs/manual/mod/directives.html.en	18 Jul 2003 15:56:57 -0000	1.6
@@ -228,6 +228,8 @@
 
       <li><a href="core.html#defaulttype">DefaultType</a></li>
 
+      <li><a href="mod_define.html#define">Define</a>
+
       <li><a href="mod_access.html#deny">Deny</a></li>
 
       <li><a href="core.html#directory">&lt;Directory&gt;</a></li>
